spec/spec_helper.rb

```diff
@@ -0,0 +1,62 @@
+# encoding: utf-8
+
+require "logstash/devutils/rspec/spec_helper"
+
+class TracerBase
+  def initialize() @tracer = []; end
+
+  def trace_for(symbol)
+    params = @tracer.map {|k,v| k == symbol ? v : nil}.compact
+    params.empty? ? false : params
+  end
+
+  def clear()
+    @tracer.clear()
+  end
+end
+
+class FileLogTracer < TracerBase
+  def warn(*args) @tracer.push [:warn, args]; end
+  def error(*args) @tracer.push [:error, args]; end
+  def debug(*args) @tracer.push [:debug, args]; end
+  def info(*args) @tracer.push [:info, args]; end
+
+  def info?() true; end
+  def debug?() true; end
+  def warn?() true; end
+  def error?() true; end
+end
+
+class ComponentTracer < TracerBase
+  def accept(*args) @tracer.push [:accept, args]; end
+  def deliver(*args) @tracer.push [:deliver, args]; end
+end
+
+class CodecTracer < TracerBase
+  def decode_accept(ctx, data, listener)
+    @tracer.push [:decode_accept, [ctx, data]]
+    listener.process(ctx, {"message" => data})
+  end
+  def accept(listener)
+    @tracer.push [:accept, true]
+  end
+  def auto_flush()
+    @tracer.push [:auto_flush, true]
+  end
+  def close
+    @tracer.push [:close, true]
+  end
+  def clone
+    self.class.new
+  end
+end
```

Comment:
I like your way of using tracer/spy classes in your tests, might be worth sharing this pattern.